modules/livestore/instance_search.go

```diff
@@ -251,6 +246,9 @@ func (i *instance) Search(ctx context.Context, req *tempopb.SearchRequest) (*tem
 		return nil
 	}
 
+	i.blocksMtx.RLock()
```

Comment:
we should leave this in `iterateBlocks()` and just remove the call in `QueryRange`. are you saying the livestore still locks with that change?

Reply:
+1, I think it's simpler code if locking is in `iterateBlocks()`